geode-core/src/distributedTest/java/org/apache/geode/internal/cache/tier/sockets/DurableClientStatsDUnitTest.java

```diff
@@ -272,9 +273,12 @@ public static void checkStatisticsWithExpectedValues(int reconnectionCount, int
       logger.info("Stats:" + "\nDurableReconnectionCount:" + stats.get_durableReconnectionCount()
           + "\nQueueDroppedCount" + stats.get_queueDroppedCount()
           + "\nEventsEnqueuedWhileClientAwayCount" + stats.get_eventEnqueuedWhileClientAwayCount());
-      assertThat(stats.get_durableReconnectionCount()).isEqualTo(reconnectionCount);
-      assertThat(stats.get_queueDroppedCount()).isEqualTo(queueDropCount);
-      assertThat(stats.get_eventEnqueuedWhileClientAwayCount()).isEqualTo(enqueueCount);
+      await().untilAsserted(
+          () -> assertThat(stats.get_durableReconnectionCount()).isEqualTo(reconnectionCount));
+      await()
+          .untilAsserted(() -> assertThat(stats.get_queueDroppedCount()).isEqualTo(queueDropCount));
+      await().untilAsserted(
+          () -> assertThat(stats.get_eventEnqueuedWhileClientAwayCount()).isEqualTo(enqueueCount));
     } catch (Exception e) {
       fail("Exception thrown while executing checkStatisticsWithExpectedValues()", e);
```

Comment:
I am a little hesitant on this `fail`. I assume it was here to bubble up some proper task failure handling but the use of `fail` still smells funny. First off this `catch` block is going to catch that assertion failures and wrap them, which makes me curious about behavior on failure.  We don't consistently catch exceptions and convert to failure, so maybe it isn't necessary. If it is I wonder if using the `assertThatCode(...).doesNotThrowAnyException()` is more appropriate replacement?